main.go

```diff
@@ -367,13 +367,19 @@ func setupControllers(ctx context.Context, mgr ctrl.Manager, sw *watch.Controlle
 		args = append(args, rego.AddExternalDataProviderCache(providerCache))
 		mutationOpts.ProviderCache = providerCache
 
-		if *externaldataProviderResponseCacheTTL <= 0 {
+		// Enable the external data provider response cache only if TTL is not 0
+		// For details, see https://github.com/open-policy-agent/gatekeeper/issues/3024
```

Comment:
I'm usually a fan of comments that point to a wider context but in this case the GH issue doesn't contain more context as to why we are adding this. Usually this would be helpful if there was a tricky, meaty problem that might've required a compromise or sub-optimal design, etc.

In this case, I think we can do without these. Wdyt?
```suggestion
```

Reply:
Thanks for the review @acpana. I have addressed your comments. PTAL.